src/misc/lv_style.h

```diff
@@ -249,11 +263,13 @@ typedef struct {
     union {
         lv_style_value_t value1;
         uint8_t * values_and_props;
+        const lv_style_const_prop_t * const_props;
```

Comment:
it's better to use the same layout for both const and non-const style element.

Reply:
Let's discuss it further.

The reason I opted for this approach was to make the syntax cleaner. Right now a constant style is simply an array of property-value pairs, which can be expressed pretty concisely in C:

```c
const lv_style_const_prop_t test_const_style_props[] = {
    LV_STYLE_CONST_BG_COLOR(LV_COLOR_MAKE(0xFF, 0, 0))
};

LV_STYLE_CONST_INIT(test_const_style, test_const_style_props);
```

Now `&test_const_style` can be used just like a normal style.

I haven't completely deciphered `values_and_props` but it seems that it does quite a bit of [manual memory management](https://github.com/lvgl/lvgl/blob/227afda49270bc5b2e7bc5f0c9097af31e8c6a0c/src/misc/lv_style.c#L153-L166), whereas the logic for reading a constant property is [much simpler](https://github.com/lvgl/lvgl/blob/a44b4e762e20df2a43a47d2447fef96c49478e8e/src/misc/lv_style.h#L353-L360). Additionally, since `const_props` is in a union with the other values, it shouldn't consume any additional flash.

Reply:
`values_and_props` stores the data like this:
```
bg_color's value (4 bytes)
bg_opa's value (4 bytes)
border_opa's value (4 bytes)
LV_STYLE_BG_COLOR  (2 bytes)
LV_STYLE_BG_OPA (2 bytes)
LV_STYLE_BORDER_OPA (2 bytes)
```
This was the data can be packed more thigtly. 

So it's not an ordinary array. Adding a different reading method in  `lv_style_get_prop` is perfect.

Reply:
@embeddedt we can use the new layout(lv_style_const_prop_t)  for the non-const case too.
BTW, If the size(the samll padding) is so critical, we can add pack to the struct.

Reply:
I think @kisvegabor's method gives better performance, as both the key and value are naturally aligned (style properties are 16-bit and values are 32-bit), but there isn't any extra memory cost as they are effectively stored separately. With a packed structure, the values will be unaligned half the time, which will probably affect performance on many chips.

Reply:
Besides what @embeddedt mentioned `__attribute__((packed))` is compiler specific.